src/main/java/seedu/address/logic/commands/eventcommands/VoteCommand.java

```diff
@@ -49,19 +49,18 @@ public VoteCommand(Index pollIndex, String optionName) {
     @Override
     public CommandResult execute(Model model, CommandHistory history) throws CommandException {
         try {
-            Event event = model.getSelectedEvent();
-            Person person = model.getCurrentUser();
-            event.addVoteToPoll(pollIndex, person, optionName);
+            Poll poll = model.voteOption(pollIndex, optionName);
             model.commitAddressBook();
-            model.updateEvent(event, event);
             String result = String.format(MESSAGE_SUCCESS, optionName, pollIndex.getOneBased());
-            String pollDisplayResult = event.displayPoll(pollIndex);
+            String pollDisplayResult = poll.displayPoll();
             EventsCenter.getInstance().post(new DisplayPollEvent(pollDisplayResult));
             return new CommandResult(result);
         } catch (IndexOutOfBoundsException e) {
             throw new CommandException(Messages.MESSAGE_NO_POLL_AT_INDEX);
         } catch (IllegalArgumentException e) {
             throw new CommandException(Messages.MESSAGE_NO_SUCH_OPTION);
+        } catch (DuplicatePersonException e) {
```

Comment:
According to Coveralls, this DuplicatePersonException was not covered by Junit test. Could you check if this is the case? And in general, coveralls has pointed out that there are quite a few files that aren't fully covered by Junit tests. Could you make sure to remedy them as well?